reflection_refraction.py

```diff
@@ -0,0 +1,184 @@
+from distutils.util import strtobool
+import math
+from matplotlib import pyplot as plt
+import typing
```

Comment:
If we use python 3.9+ the way you have written this won't need the import of `typing` and you can remove this. 

In lower versions of python one would need `from typing import Tuple` whereas in python 3.9+ you can directly use the build in type `tuple` as you are already doing. 

Check out [this discussion](https://stackoverflow.com/a/62033243) as well.